packages/core/contracts/financial-templates/perpetual-multiparty/ConfigStore.sol

```diff
@@ -0,0 +1,148 @@
+        uint256 updateLiveness;
+        // Reward rate paid to successful proposers. Percentage of 1 E.g., .1 is 10%.
+        FixedPoint.Unsigned rewardRatePerSecond;
+        // Bond % (of given contract's PfC) that must be staked by proposers. Percentage of 1, e.g. 0.0005 is 0.05%.
+        FixedPoint.Unsigned proposerBondPct;
+    }
+
+    // Make currentConfig private to force user to call getCurrentConfig, which returns the pendingConfig
+    // if its liveness has expired.
+    ConfigSettings private currentConfig;
+
+    // Beginning on `pendingPassedTimestamp`, the `pendingConfig` can be published as the current config.
+    ConfigSettings public pendingConfig;
+    uint256 public pendingPassedTimestamp;
+
+    /****************************************
+     *                EVENTS                *
+     ****************************************/
+
+    event ProposedNewConfigSettings(
+        address indexed proposer,
+        uint256 rewardRate,
+        uint256 proposerBond,
+        uint256 updateLiveness,
+        uint256 proposalPassedTimestamp
+    );
+    event ChangedNewConfigSettings(uint256 rewardRate, uint256 proposerBond, uint256 updateLiveness);
+
+    /****************************************
+     *                MODIFIERS             *
+     ****************************************/
+
+    modifier updateConfig() {
+        // Update config settings if possible.
+        _updateConfig();
+        _;
+    }
+
+    constructor(ConfigSettings memory _initialConfig, address _timerAddress) public Testable(_timerAddress) {
+        _validateConfig(_initialConfig);
+        currentConfig = _initialConfig;
+    }
+
+    /**
+     * @notice Returns current config or pending config if pending liveness has expired.
+     * @return ConfigSettings config settings that calling financial contract should view as "live".
+     */
+    function getCurrentConfig() external view override nonReentrantView() returns (ConfigSettings memory) {
+        if (_pendingProposalPassed()) {
+            return pendingConfig;
+        } else {
+            return currentConfig;
+        }
+    }
+
+    /**
+     * @notice Propose new configuration settings. New settings go into effect
+     * after a liveness period passes.
+     * @dev Callable only by owner. Calling this while there is already a pending proposal
+     * will overwrite the pending proposal.
+     */
+    function proposeNewConfig(ConfigSettings memory newConfig) external onlyOwner() nonReentrant() updateConfig() {
+        // Alternatively, we can revert if there is already a pending proposal.
+        // require(pendingPassedTimestamp == 0, "Pending proposal exists");
+
+        _validateConfig(newConfig);
+
+        // Warning: This overwrites a pending proposal!
+        pendingConfig = newConfig;
+
+        // Use current config's liveness period to timelock this proposal.
+        pendingPassedTimestamp = getCurrentTime() + currentConfig.updateLiveness;
```

Comment:
Is this liveness only used for this? Or is this meant to control liveness periods inside the contract as well?

Reply:
Yes its only meant to control the timelock for this ConfigStore contract

Reply:
I can rename it to `configStoreTimelock` for more accuracy

Reply:
Yeah, I think `timelockLiveness` or `configStoreTimelock` would make that more obvious. +1

Reply:
Done with `timelockLiveness`